src/transformers/feature_extraction_utils.py

```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+ Feature extraction classes for python tokenizers.
+"""
+
+
+class BatchFeature(UserDict):
+    """"""
+
+    def __init__(
+        self,
+        data: Optional[Dict[str, Any]] = None,
+        encoding: Optional[Union[EncodingFast, Sequence[EncodingFast]]] = None,
+        tensor_type: Union[None, str, TensorType] = None,
+        prepend_batch_axis: bool = False,
+        n_sequences: Optional[int] = None,
+    ):
+        super().__init__(data)
+        # add similar functionality as BatchEncoding
+
+
+class PreTrainedFeatureExtractor:
```

Comment:
Nice! Should the `__call__` method be already defined in the superclass? It looks like it could be general enough that it wouldn't need to be overridden in children classes, but maybe we'll only see that once we have a few models implemented and we can refactor.

What other methods can we expect from the `PreTrainedFeatureExtractor`, methods like truncating as an addition to padding?

Reply:
Since, the feature processing can be quite different depending on whether:

- raw waveform features,
- short time fourier transform bin,
- Log-mel, or 
- Cepstrum features

are used so I would put the `__call__` into the respective feature processors.

Besides padding I think we could put methods like `normalization` and `from_file` class methods in this `PreTrainedClass`

Reply:
I love the design! Regarding the padding and truncation, it can be different for different models, for example in S2T we need to pad the log_mel length dimension. Can it be handled in the generic `pad` method?